client/src/Categories/CategoriesSources.js

```diff
@@ -0,0 +1,40 @@
+import { Link } from 'react-router-dom';
+import { useParams } from 'react-router-dom';
+
+import './CategoriesList.scss';
+import { useEffect, useState } from 'react';
+import Api from '../Api';
+
+function CategoriesList() {
+  const { slug } = useParams();
+  const [category, setCategory] = useState([]);
+  const [resources, setResources] = useState([]);
+
+  useEffect(
+    function () {
+      if (slug) {
+        Api.categories.getSlug(slug).then((response) => setCategory(response.data));
+        Api.resources.index(category.id).then((response) => setResources(response.data));
```

Comment:
I see what you're doing here now... but it shouldn't be necessary... we should be able to use the id and slug interchangeably, so that I don't think we need a new getSlug(), just the standard get() with the slug passed in should work...

Reply:
Also, beycause the API functions are _asynchronous_ they are executed one after each other even before the response is received.

Since resources.index() is referencing the category, I don't think this works- category is an empty array to start...

This won't be a problem if we switch to using the slug in the resources.index() call...

Reply:
I have fixed this to use `get(slug)`.

Reply:
You're still passing category.id into Api.resources.index() which isn't going to work...

Instead, just pass the slug to that call, i.e. `Api.resources.index(slug)...`

Then, in `routes/api/resources.js` you can handle it and filter the resources returned...